File: leetcode/2343.裁剪数字后查询第 K 小的数字.cpp
class Solution {
public:
    vector<int> smallestTrimmedNumbers(vector<string>& nums, vector<vector<int>>& queries) {
        int m = nums.size(), n = queries.size();
        vector<int> answer;
        for (int i = 0; i < n; i++) {
            vector<pair<string,int>> num_c;
            for (int j = 0; j < m; j++) {
                string a =nums[j].substr(nums[j].size()-queries[i][1], nums[j].size());
                num_c.push_back({ a,j });
            }
            stable_sort(num_c.begin(),num_c.end());
            answer.push_back(num_c[queries[i][0] - 1].second);
        }
        return answer;
    }
};

File: leetcode/2343.裁剪数字后查询第 K 小的数字.py
class Solution:
    def smallestTrimmedNumbers(self, nums: List[str], queries: List[List[int]]) -> List[int]:
        m,n=len(nums),len(queries)
        answer=[]
        for i in range(n):
            arr=[]
            for j in range(m):
                k=queries[i][1]
                arr.append([nums[j][-k:],j])
            arr.sort()
            answer.append(arr[queries[i][0]-1][1])
        return answer
                